### ISSUE TITLE
Multiple IP Addresses per Profile

### ISSUE THREAD

**Issue by KurtisLiggett** [KurtisLiggett/Simple-IP-Config#8]
Allow option to add multiple IP addresses to each profile.
-Similar to 'Advanced...' button in Windows' built-in configuration screen.

**Comment by giacomocerquone** [comment]
Sorry but I can't understand this... multiple ip address per profile shouldn't/couldn't be stored! What do you really mean?

**Comment by KurtisLiggett** [comment]
**I really mean multiple addresses per profile!**

Here is my vision for the GUI:
The GUI would be switched to a new mode showing listviews for the IP/DNS addresses. The columns would be edited by clicking on the address you want to edit in-place. It would have a add/delete buttons to add addresses.
I will post a drawing once I have time.

The ini file will store addresses like this
IpAddress=192.168.1.1|10.225.251.12
IpSubnet=255.255.255.0|255.255.255.0

It would be similar to the built-in Windows functionality, but easier to use.
Something like this:
![image](https://user-images.githubusercontent.com/29669372/28476515-e02f1b68-6e15-11e7-99ee-5b9cc1298b7c.png)

**Comment by giacomocerquone** [comment]
Ok but this doesn't make sense to me... I mean when you apply a profile with multiple IP addresses which one will be applied to the OS? And secondly, to me it seems completely superfluous since you can just create multiple profiles with different ips

**Comment by KurtisLiggett** [comment]
I think you misunderstand. Here is my terrible explanation.
Sometimes it is necessary to assign 2 or more IP addresses to a single network adapter. This allows you to be on 2 networks at the same time.
For example: you can have an IP of 192.168.1.1 and 10.225.251.1 at the same time without having to change your IP back and forth.

**Comment by giacomocerquone** [comment]
Oh... My huge ignorance was telling me that being on two networks contemporary  wasn't possible! So sorry haha anyway this is a very interesting feature. After the code-cleanup I could help you here

**Comment by youRFate** [comment]
Yes pleaaase. That is the reason I downloaded this tool, as I often have to change the different IP addresses one of the network adapters on this machine has. You will have my eternal gratitiude if you allow me do easily add / remove ip addreses in a nice interface with some overview. Also, please allow for the / notation of the subnet mask, for example 192.168.0.70/24.

**Comment by Otiel** [comment]
I would love to see this feature as well.

**Comment by KurtisLiggett** [comment]
I've started working on the backend of this. Just need to make a nice GUI for it.

If I see a lot of interest, I'll put more effort into getting this done sooner.

**Comment by saau** [comment]
I would like love to see this feature as well.

**Comment by KurtisLiggett** [comment]
_From @raym0nde commented on Nov 30, 2017_
_Copied from duplicate issue #58._

I have an application that requires two IP addresses on a single adapter. It would be great if SIC had the ability to set multiple IPs.


![image](https://user-images.githubusercontent.com/24615894/33429990-ddb8b99a-d59c-11e7-9d15-c81059bce09a.png)

**Comment by madbanzai** [comment]
Any update on this feature? It would be great to see it soon.

**Comment by White-Goblin** [comment]
I'm also requesting the ability to assign multiple IP addresses to a single NIC. I use software which requires this.

**Comment by quarky42** [comment]
I see now that there was already a thread about multiple IPs on a single NIC.  I should have posted here.

My main question:  As a shortcut, could this be implemented without the GUI support at first?   Example:  If we know the proper way to edit the INI file, could we go ahead and take an existing profile and add the second IP and second Subnet to it?   (If we knew the delimeter, say a comma or something, we could just add that delimeter, add the second entry and save the file.)

The GUI would need to be able to not barf on having an unusual entry there, but if the backend code could just read the profile.ini, see that there are two IPs for that interface and assign them to the selected NIC, that would be great.   Ultimately, being able to edit the IPs in the interface would be preferred and having something similar to the Windows interface for that would be great, but the GUI is a nice to have where as the ability to assign two IPs to the NIC would be a great to have feature for me.

Without redesigning the GUI completely, one could just simply add an "Add" / "Del" buttons, some navigation arrows, and a visual indicator of how many IPs are currently being stored for that one profile.     Clicking the Add button would save the current screen, increment the index/counter, provide blank text entry boxes, and the user could then enter the next values.    The Del button could delete the currently displayed values.  The navigation arrows would update the IP and Subnet fields to show the values associated with the currently selected index/counter.    It's not as pretty as a list, but maybe easier to just add a handful of buttons to get this feature out the door?

**Comment by hurr1k4ne** [comment]
![2020-04-21 11_59_08-Window](https://user-images.githubusercontent.com/3234219/79852957-b2388300-83c7-11ea-8b8b-cf02cb4c0338.png)

Same Problem here.
At my work i need the .125. Network to reach the Gateway and Attendees Network.
In addition i need the .120. Network to get access to the business administrations network. 

SECURITY by Complexity....

INI profile edit style like @KurtisLiggett is more than sufficent for me.
